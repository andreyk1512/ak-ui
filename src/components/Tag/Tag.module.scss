@import "vars";

.tag {
  padding: 7px 10px;
  border-radius: 3px;
  letter-spacing: 0.36px;
  line-height: 100%;
  font-weight: $font-weight-medium;
  font-size: $font-size-micro;

  .blue {
    color: $carolina-blue;
    background-color: $light-sky-blue-40;
  }

  .green {
    color: $light-sea-green;
    background-color: $light-sea-green-20;
  }

  .orange {
    color: $tigers-eye;
    background-color: $tigers-eye-20;
  }

  .purple {
    color: $amethyst;
    background-color: $amethyst-20;
  }

  .pink {
    color: $telemagenta;
    background-color: $telemagenta-20;
  }
}