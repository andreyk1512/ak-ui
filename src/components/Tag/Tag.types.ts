import { ReactNode } from 'react';

export type TagColor = 'blue' | 'green' | 'orange' | 'purple' | 'pink';

export type TagProps = {
  children: ReactNode;
  color: TagColor;
};
