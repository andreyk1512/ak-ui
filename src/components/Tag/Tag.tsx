import { css, SerializedStyles } from '@emotion/react';
import * as styles from '../../styles';
import { TagProps } from './Tag.types';

const colorsMap: Record<TagProps['color'], SerializedStyles> = {
  blue: css`color: ${styles.color.carolinaBlue}; background-color: ${styles.color.lightSkyBlue_40};`,
  green: css`color: ${styles.color.lightSeaGreen}; background-color: ${styles.color.lightSeaGreen_20};`,
  orange: css`color: ${styles.color.tigersEye}; background-color: ${styles.color.tigersEye_20};`,
  pink: css`color: ${styles.color.amethyst}; background-color: ${styles.color.amethyst_20};`,
  purple: css`color: ${styles.color.telemagenta}; background-color: ${styles.color.telemagenta_20};`
};

export const Tag = ({ children, color }: TagProps) => {
  const tagStyle = css`
    padding: 7px 10px;
    border-radius: 3px;
    letter-spacing: 0.36px;
    line-height: 100%;
    font-size: ${styles.fontSize.micro};
    font-weight: ${styles.fontWeight.medium};
    ${colorsMap[color]}
  `;

  return (
    <span css={tagStyle}>{children}</span>
  );
};
