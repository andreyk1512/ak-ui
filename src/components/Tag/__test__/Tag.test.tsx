import { render } from '@testing-library/react';
import { Tag } from '../Tag';

describe('Tag', () => {
  it('should correctly render span element with content', () => {
    const children = 'ONLINE';
    const { container } = render(<Tag color="green">{children}</Tag>);

    expect(container.querySelector('span')?.textContent).toBe(children);
    expect(container.firstChild).toMatchSnapshot();
  });
});
