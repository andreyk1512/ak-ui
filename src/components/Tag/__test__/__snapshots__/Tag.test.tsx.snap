// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should render the tag text 1`] = `
<span
  class="css-1mp94v1-Tag"
>
  My Tag
</span>
`;
